frontend/src/router/pages/PreviewChanges.jsx

```diff
@@ -82,7 +109,10 @@ function PreviewChanges() {
     });
     setNotes("");
     fetchData();
-    setShowFlash(true);
+    openFlashMessage(
+      "Dates saved as draft",
+      `${data?.park.name} ${data?.featureType.name} ${data?.operatingYear} season details saved`,
```

Comment:
I suppose `data` will always be fully populated by the time this is showing anyway, but FYI the `?.` optional chaining operator doesn't "short circuit" the whole chain so in cases like this you'd have to put them at every `.` in the chain like this:

```suggestion
      `${data?.park?.name} ${data?.featureType?.name} ${data?.operatingYear} season details saved`,
```

Otherwise if `park` is null or undefined, it will try to read ".name" from it and throw a TypeError.

Reply:
I guess I didn't think this was necessary because, park and featureType should always be there. If park or featureType are not part of the data nothing else should work either and we should expect to see an error.

Reply:
If data is always set in this context, then we don't need any `?`s. All or nothin', I guess 👍

Reply:
It could happen that data is null because it depends on the hook to fetch the data. But, once we know that `data` is there, park and featureType should always be there. So, I think it makes sense to just have the `?` after `data`. This is consistent with how it's done in other places of the app.  
I can just add all the ? after every object if you like, but I don't think it would make a difference.
I found a weird state that happens only rare occasions in which loading is false, error is null, and data is null. So, I don't think we can get rid of all of the `?` until we figure that out.

Reply:
I just mean that the question mark before "park" wouldn't help if `data` was null because of the way optional chaining works. You can try something like this in the console:

<img width="527" alt="image" src="https://github.com/user-attachments/assets/542138bb-52a9-4590-859e-1d9895639cdd" />

So as long as we guard that chunk of the template to not render until `data` is populated, we can ditch the question marks. Or we can add them all the way through the chain, but having it only part way would cause problems because it doesn't "short circuit" and it keeps checking all he way down the chain.